drivers/afe/ad4110/ad4110.c

```diff
@@ -777,6 +795,45 @@ int32_t ad4110_setup(struct ad4110_dev **device,
 	return ret;
 }
 
+/***************************************************************************//**
+ * ADC continuous read fills buffer with buffer_size number of samples.
+ *
+ * @param dev          - The device structure.
+ * @param buffer       - The buffer.
+ * @param buffer_size  - The buffer size = number of samples.
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t ad4110_continuous_read(struct ad4110_dev *dev, int32_t *buffer,
+			       int32_t buffer_size)
+{
+	int32_t ret;
+
+	struct ad4110_callback_ctx ctx = {
+		.dev = dev,
+		.buffer = buffer,
+		.buffer_size = buffer_size,
+	};
+
+	struct callback_desc irq_callback = {
+		.callback = &irq_adc_read,
+		.ctx = &ctx
+	};
+
+	ret = irq_trigger_level_set(dev->irq_desc, dev->nready_pin, IRQ_LEVEL_LOW);
+	ret |= irq_register_callback(dev->irq_desc, dev->nready_pin, &irq_callback);
+
+
+	ret |= ad4110_set_adc_mode(dev, AD4110_CONTINOUS_CONV_MODE);
+	mdelay(1U);
+	ret |= irq_enable(dev->irq_desc, dev->nready_pin);
+
+	while(ctx.buffer_size != 0U)
+		;
```

Comment:
`while(ctx.buffer_size != 0U);` ?

Reply:
Function waits for the interrupts to occur and the buffer to be filled. (by the interrupts)

Reply:
I was suggesting to put `;` and the while statement in one line

Reply:
Done